config/resolve_credentials_test.go

```diff
@@ -0,0 +1,222 @@
+package config
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"reflect"
+	"runtime"
+	"testing"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/internal/awstesting"
+	"github.com/aws/aws-sdk-go-v2/service/sts"
+	"github.com/awslabs/smithy-go/middleware"
+	smithyhttp "github.com/awslabs/smithy-go/transport/http"
+)
+
+func TestSharedConfigCredentialSource(t *testing.T) {
+	const configFileForWindows = "testdata/credential_source_config_for_windows"
+	const configFile = "testdata/credential_source_config"
```

Comment:
nit these paths aren't compatible with windows should use `filepath.Join` instead.